inference-engine/ie_bridges/c/src/ie_c_api.cpp

```diff
@@ -335,6 +336,64 @@ IEStatusCode ie_core_read_network_from_memory(ie_core_t *core, const uint8_t *xm
     return status;
 }
 
+IEStatusCode ie_core_import_network(ie_core_t *core, const char *file_name, const char *device_name, \
+        const ie_config_t *config, ie_executable_network_t **exe_network) {
+    IEStatusCode status = IEStatusCode::OK;
+
+    if (core == nullptr || file_name == nullptr || device_name == nullptr || config == nullptr || exe_network == nullptr) {
+        status = IEStatusCode::GENERAL_ERROR;
+        return status;
+    }
+
+    try {
+        std::map<std::string, std::string> conf_map = config2Map(config);
+        std::unique_ptr<ie_executable_network_t> exe_net(new ie_executable_network_t);
+
+        exe_net->object = core->object.ImportNetwork(file_name, device_name, conf_map);
+        *exe_network = exe_net.release();
+    } CATCH_IE_EXCEPTIONS
+
+    return status;
+}
+
+IEStatusCode ie_core_import_network_from_memory(ie_core_t *core, const ie_blob_t *model_blob, const char *device_name, \
+    const ie_config_t *config, ie_executable_network_t **exe_network) {
+    if (core == nullptr || model_blob == nullptr ||device_name == nullptr || config == nullptr || exe_network == nullptr) {
+        return IEStatusCode::GENERAL_ERROR;
+    }
+
+    IEStatusCode status = IEStatusCode::OK;
+    try {
+        IE::MemoryBlob::Ptr mblob = IE::as<IE::MemoryBlob>(model_blob->object);
+        char *blob_data = mblob->rmap().as<char *>();
+
+        size_t blob_size = model_blob->object->byteSize();
+        std::stringstream network_model(std::string(blob_data, blob_size));
```

Comment:
Seems a huge memory copy can be done here. What about using https://stackoverflow.com/questions/13059091/creating-an-input-stream-from-constant-memory

Reply:
@apankratovantonp  thanks for the hit, a bit changed and implemented in the last commit.